src/app/middleware/verifysignup.js

```diff
@@ -0,0 +1,13 @@
+import UserServices from '../../database/acid/services/userServices';
+import GenericHandler from '../helpers/responses';
+
+const checkUsername = async (req, res, next) => {
+  const { username } = req.body;
+  const usernameExist = await UserServices.findUsername({ username });
+  if (usernameExist) {
+    return GenericHandler.error(400, 'Username already in use!', res);
```

Comment:
I can suggest you to create a file where you store your status code and their messages. then import them to make clean your code where you want to use them. E.g : `export const HTTP_CREATED_MESSAGE = 'User registered successfully';` and in the other file you can create different status code: `export const HTTP_OK = 200;`. But the rest looks good. Nice work